src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -369,6 +369,14 @@
         "LIMIT", "GROUP", "ORDER", "HAVING", "QUALIFY", "WINDOW"
     ),
     PrimaryKeyGrammar=Sequence("PRIMARY", "KEY"),
+    # Odd syntax, but prevents eager parameters being confused for data types
+    FunctionParameterGrammar=OneOf(
```

Comment:
Extracted this from the existing `FunctionParameter` grammar to minimize duplication in BigQuery dialect